eth/accountmanager.go

```diff
@@ -136,12 +137,13 @@ func (am *accountManager) CreateTransactOpts(gasLimit uint64, gasPrice *big.Int)
 
 // Sign a transaction. Account must be unlocked
 func (am *accountManager) SignTx(signer types.Signer, tx *types.Transaction) (*types.Transaction, error) {
```

Comment:
Omit the `signer` argument?

In the discussion we had earlier this week I mentioned ignoring a `signer` argument, but just to clarify I meant the `signer` argument in [bind.TransactOpts.Signer](https://github.com/livepeer/go-livepeer/blob/master/eth/accountmanager.go#L127).

Reply:
I know but wouldn't that obfuscate the possibility for us to add custom signing logic. 

Since every signing actual will use that function band which calls `am.SignTx` it seems reasonable to implement it there while still giving us the option to implement addition logic on top in the `accountManager` if we deem it necessary without a huge refactor.

Reply:
> Since every contract transaction to be signed will use that binding that calls am.SignTx() anyhow it seemed reasonable to implement it there instead since the binding is just a pass-through essentially to a method on our own accountManager implementation.

We also use `SignTx()` in [ReplaceTransaction()](https://github.com/livepeer/go-livepeer/blob/master/eth/client.go#L799) so having it around still has some use.

Reply:
Not sure if we're on the same page , so it's okay as is? 

Or do you want to change `	SignTx(signer types.Signer, tx *types.Transaction) (*types.Transaction, error)` into `	SignTx(tx *types.Transaction) (*types.Transaction, error)` ?

Reply:
I don't see the utility of having the `signer` param in `SignTx()` if `AccountManager` is always going to be configured with a specific `Signer` so I think omitting the `signer` param would make the API clearer especially since `SignTx()` is also used in `ReplaceTransaction()`.

Reply:
gotcha, fixed in f297f49a